fhir-persistence-jdbc/src/main/java/com/ibm/fhir/persistence/jdbc/dao/impl/ParameterVisitorBatchDAO.java

```diff
@@ -708,4 +745,136 @@ private void closeStatement(PreparedStatement ps) {
     private boolean isBase(ExtractedParameterValue param) {
         return "Resource".equals(param.getBase());
     }
-}
+    
+    /**
+     * Get the leading part of the url e.g. https://example.com
+     * @return
+     */
+    private String getServerUrl() throws FHIRPersistenceException {
+        
+        if (this.serverBase != null) {
+            return this.serverBase;
+        }
+        
+        String uri = FHIRRequestContext.get().getOriginalRequestUri();
+
+        // request URI is not set for all unit-tests, so we need to take that into account
+        if (uri == null) {
+            return null;
+        }
+        
+        try {
+            StringBuilder result = new StringBuilder();
+            URL url = new URL(uri);
+            
+            result.append(url.getProtocol());
+            result.append("://");
+            result.append(url.getHost());
+            
+            if (url.getPort() != -1) {
+                result.append(":");
+                result.append(url.getPort());
+            }
+            
+            // https://example.com:9443/
+            result.append("/");
+            
+            final String endpoint = "fhir-server/api/v4/";
```

Comment:
```suggestion
            final String endpoint = "fhir-server/api/v4/";
```

This is not always true. 

Consider our interop enviroment which has dev-1234

Reply:
Reading below, I'm not even sure this is needed

Reply:
Where can we get this value from?

Reply:
Raised issue #1635 to address this after.